package/libpiksi/libpiksi/include/libpiksi/metrics.h

```diff
@@ -56,13 +56,16 @@ typedef struct {
   u64 ns;
 } pk_metrics_time_t;
 
-typedef union pk_metrics_value_s {
-  s32 s32;
-  s64 s64;
-  u32 u32;
-  u64 u64;
-  double f64;
-  pk_metrics_time_t time;
+typedef struct pk_metrics_value_s {
+  pk_metrics_type_t type;
+  union {
+    s32 s32;
```

Comment:
Is this a good idea to have variable name shadowing the variable name?

Reply:
Nothing obviously great comes to my mind either. Maybe `value_s32`, `value_s64` ...?

Reply:
I expanded `.d` to `.data` and adding a `.as_` prefix to all the union members-- I think this makes sense since the union is like a form of casting and it makes the struct deconstructions read a lot nicer `update.value.data.as_u32` instead of `update.value.d.u32`.